### PR TITLE
 #216: Updated dependencies to fix vulnerabilities and refactorings

### LINKED ISSUE #216: Fix vulnerabilities and refactor some parts
* [x] Use latest drivers and ExaPlus from the URLS behind [downloads.exasol.com](https://downloads.exasol.com/clients-and-drivers/odbc)
  * Old: https://exasol-script-languages-dependencies.s3.eu-central-1.amazonaws.com/EXASOL_ODBC-7.0.11.tar.gz
  * New: https://x-up.s3.amazonaws.com/7.x/24.1.1/Exasol_ODBC-24.1.1-Linux_x86_64.tar.gz
* [x] Use latest version of exasol-python-test-framework (EPTF)
  * incl. feature https://github.com/exasol/exasol-python-test-framework/issues/66
* [ ] Add CLI option `FINGERPRINT=NoCertCheck` for Exasol DB &ge; version 8
* [x] Fix vulnerabilities 
  * CVE-2024-35195 in dependency `requests` in versions below `2.32.0` caused by requests `Session` object not verifying requests after making first request with `verify=False`
  * CVE-2024-37891 in transitive dependency via `boto3` to `urllib3` in versions below `2.2.2` caused by proxy-authorization request header not to be stripped during cross-origin redirects as no update of notebook-connector is available, yet.
  * GHSA-w235-7p84-xx57 in transitive dependency via `luigi` to `tornado` in versions below `6.4.1` enabling CRLF injection in `CurlAsyncHTTPClient` headers.
  * GHSA-753j-mpmx-qq6g in transitive dependency via `luigi` to `tornado` in versions below `6.4.1` due to inconsistent interpretation of HTTP Requests ('HTTP Request/Response Smuggling')
* [x] Ignore vulnerabilities 
  * GHSA-753j-mpmx-qq6g in dependency `configobj` in versions <= `5.0.8` being ReDoS exploitable by developers using values in a server-side configuration file as SLCT is used only client side and pathed version are not available, yet.

### PR DESCRIPTION
**Pull request by ckunki**
Closes #216

### CODE REVIEW

File: doc/dependencies.md
Comment (kaklakariada) [inline doc/dependencies.md:1]: Is this generated? If yes, please add to gitattributes

File: exasol_script_languages_container_tool/lib/tasks/test/run_db_test.py [hunk 1]
```diff
@@ -36,11 +37,27 @@ def __init__(self, *args, **kwargs):
...
+    def _run_command(
+            self,
+            docker_client: docker.client,
+            container: docker.models.containers.Container,
+            command: str,
+    ) -> str:
+        file = StringIO()
+        exit_code = exec_run_and_write_to_stream(docker_client, container, command, file, {})
+        if exit_code != 0:
+            raise Exception(f"Command returned {exit_code}: {command}")
```
Comment (kaklakariada) [inline exasol_script_languages_container_tool/lib/tasks/test/run_db_test.py:49]: Linter will complain about raising a too generic exception. Better use a more specific one if possible.

**Review by kaklakariada — DISMISSED** [review]

File: exasol_script_languages_container_tool/lib/tasks/test/run_db_test.py (hunk 1 above)
Comment (ckunki) [reply exasol_script_languages_container_tool/lib/tasks/test/run_db_test.py:49]: See next push.

File: doc/dependencies.md
Comment (ckunki) [reply doc/dependencies.md:1]: Not generated.

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**